### ISSUE TITLE
Infinite loading

### ISSUE THREAD

**Issue by levouh** [stevearc/aerial.nvim#4]
Not entirely sure how to go about debugging this one, however I'd imagine that I should first try having a more minimal configuration to see if something else is conflicting, etc.

Whenever I `AerialToggle`, I am just faced with the quickfix list being opened, and then what is seemingly the _Aerial_ buffer just infinitely loading. I've tried this with _Python_, _Lua_, and _C_ language servers (all of which work just fine) and get the same thing for all three. Something like this (note that I intentionally have syntax errors to show that the language server is indeed working):

![image](https://user-images.githubusercontent.com/31262046/123117923-6932f300-d3ff-11eb-9c22-853f0563c342.png)

I've gone through the _LSP_ logs (although I wouldn't expect them to show anything), etc. and haven't found anything of interest.

How do you suggest I go about debugging this? What else can I dump here?

**Comment by stevearc** [comment]
The big clue here is that quickfix is opening. The default `documentSymbol` handler puts the symbols into quickfix and opens it https://github.com/neovim/neovim/blob/0d7e33bc1307117127e519fb12ec491a1a1ebd82/runtime/lua/vim/lsp/handlers.lua#L208-L215

This likely means that we have not successfully installed our `documentSymbol` handler, so aerial isn't getting any symbols. This is where we're adding the handler: https://github.com/stevearc/aerial.nvim/blob/755efbf47d9231cd1b31fed6d8fa180a59ae926f/lua/aerial.lua#L83
My guess is that either
1) You're not calling `require'aerial'.on_attach` (see [setup instructions](https://github.com/stevearc/aerial.nvim#setup)) or
2) You have some other plugin or config that runs after `aerial.on_attach` that replaces the `documentSymbol` handler.

Do either of those seem plausible?

----------
I started writing up debugging steps before I realized it was probably the issue above. I'm going to keep them below for reference if anyone else stumbles across this issue.

1) Until 0256b53cb372d226c18d608b8af0ab37179336ec, if there were no symbols found (or none of them passed the filter) the loading spinner would show forever. Make sure you're ahead of that commit and/or try setting `g:aerial_filter_kind = v:false`.

2) Double check that we are requesting symbols and the server is sending them. Set the log level to debug `lua vim.lsp.set_log_level('debug')` and look in the log file `lua print(vim.lsp.get_log_path())`. You should expect to see both a `client.request` and a `rpc.send.payload` for `textDocument/documentSymbol`.

**Comment by levouh** [comment]
It ended up being something on my end and the way that I was compiling (and also loading) my plugins. My `on_attach` configuration is loaded when a set of `filetype` criteria is met, which at that point in time hasn't loaded `aerial.nvim` (mostly with `packadd` as I have it as `opt`) so `local ok, aerial = pcall(require, 'aerial')` wouldn't actually work so it isn't "attached" to the buffer when it is loaded? Something along those lines.

I apologize for wasting your time as essentially all I had to do was `log.fmt_info('ok=%s, aerial=%s', ok, aerial)` to see that it wasn't _actually_ loading it at the right time even though it appeared to be working (the buffer loads, etc. even though it had the infinite loading).

Fixing the dependency order and whatnot, and here we have it:

![image](https://user-images.githubusercontent.com/31262046/123197646-0e35e600-d469-11eb-915b-62830cff3e0c.png)

The (albeit short due to my own neglegence) time I've spent with the plugin has proved it incredibly useful and a vast improvement over what I had before. Thanks for the work and help here.
